src/data/minors/ai.ts

```diff
@@ -0,0 +1,119 @@
+import { CollegeOrMajorRequirement, Course } from '../../requirements/types';
+import {
+  includesWithSingleRequirement,
+  courseMatchesCodeOptions,
+  ifCodeMatch,
+  courseMatchesCode,
+} from '../../requirements/checkers';
+import { AdvisorGroup } from '../../tools/advisors/types';
+import { lastNameRange } from '../../tools/advisors/checkers';
+
+const aiMinorRequirements: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Machine Learning',
+    description: 'Add Core Course 1',
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement(
+      'CS 3780',
+      'ECE 3200',
+      'ECE 4200',
+      'ORIE 3741',
+      'ORIE 4741',
+      'STSCI 3740'
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+
+  {
+    name: 'Reasoning',
+    description: 'Add Core Course 2',
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement('CS 3700'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Human-AI Interaction',
+    description: 'Add Core Course 3',
+    //TODO: add INFO 3450 exception for students grad in dec 2024 or may 2025
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement('INFO 4940'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Ethics, Governance & Policy',
+    description: 'Add Core Course 4',
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement('ENGRG 3605', 'INFO 1260', 'PUBPOL 4210'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'AI Electives',
```

Comment:
@nidhi-mylavarapu make sure this cross listing (bug?) does not break the functionality. 
![image](https://github.com/user-attachments/assets/5ec70ae0-d9f2-410b-8478-e06b8596161f)

Reply:
Right, this was a weird bug -- grabbing a course from the first bucket shouldn't cause the courses in the second bucket to change to their cross listed names. I would look for another major or minor that requires you to select 2/n courses to fulfill a requirement and repeat that functionality.